src/partisan_hiscamp_membership_strategy.erl

```diff
@@ -62,19 +62,20 @@ join(#hiScamp{membership=Membership0, level1=L1, level2=L2}=State0, Node, _NodeS
     {ok, Membership, OutgoingMessages, State}.
 
 try_distance(#hiScamp{membership=Membership0, distance=Dist}=_State0, Node) ->
+    Dist = 0,
     Dict = get(distance_metrics),
     try dict:find(Node, Dict) of
         {ok, Dist} -> Dist
     catch
-         error:E -> {"Error computing distance"}
+        {error, Error} -> Error
```

Comment:
This isn't going to work -- you had it right the first time if you're trying to match exceptions.  However, you're not trying to catch exceptions here.

If you see my other comment, the code you're calling is *not going to generate an exception.*  This code needs to use a case statement, not a try/catch.

Reply:
so like a case where I just check if there's an error or not?

Reply:
Yes, if you look at the docs for ```dict:find```, it returns two different types of values: ```{ok, Value}``` or ```error```.  You need to match against these and take action accordingly.  Before, your code was assuming it would return a number, because you were trying to use ```<``` on it, but neither case is a number.

Reply:
ohhhhh ok, gotcha